test-suite/src/test/scala/org/scalajs/testsuite/niobuffer/BufferAdapter.scala

```diff
@@ -0,0 +1,196 @@
+package org.scalajs.testsuite.niobuffer
+
+import java.nio._
+
+abstract class BufferAdapter[BT <: Buffer, ET] {
+  type BufferType = BT
+  type ElementType = ET
+
+  def slice(): BufferType
+  def duplicate(): BufferType
+  def asReadOnlyBuffer(): BufferType
+  def get(): ElementType
+  def put(e: ElementType): BufferType
+  def get(index: Int): ElementType
+  def put(index: Int, e: ElementType): BufferType
+  def get(dst: Array[ElementType], offset: Int, length: Int): BufferType
+  def get(dst: Array[ElementType]): BufferType
+  def put(src: BufferType): BufferType
+  def put(src: Array[ElementType], offset: Int, length: Int): BufferType
+  def put(src: Array[ElementType], dummy: Boolean = false): BufferType
```

Comment:
There is no need for a one-time type. There is one exactly for this purpose in Predef (the one I linked). IMHO this is is preferable to default arguments, since it is apparent from only the type signature (and not signature with parameter names) that the argument is a dummy.